### ISSUE TITLE
Custom log4j2 configuration not working with Neo4j 4.2.x

### ISSUE THREAD

**Issue by nioertel** [neo4j/neo4j#12655]
## Bug report
Since Neo4j 4.2.x (specifically commit 321386148360427c5f167f72821682a3648040f3 which was probably part of the changes related to #12631) custom Log4j 2 configurations don't work anymore.
This may be also related to [this Log4j 2 issue](https://issues.apache.org/jira/browse/LOG4J2-673) and [this Stack Overflow thread](https://stackoverflow.com/questions/48033792/log4j2-error-statuslogger-unrecognized-conversion-specifier), specifically the Log4j2Plugins.dat cache.
The issue is the same with Neo4j 4.2.0 and 4.2.1.

**Neo4j Version:** 4.2.x
**Operating System:** macOS 10.15.7 (but affects also Linux; not tested on Windows)
**API:** Neo4j Embedded (also affects Neo4j Docker Image)

### Steps to reproduce
1. Create simple project with dependencies `com.neo4j:neo4j-enterprise` and  `org.apache.logging.log4j:log4j-core`
2. Put a custom log4j2-test.json in src/test/resources
3. Create a unit test that uses a log4j logger and prints a message
4. Run the unit test

### Expected behavior
The log message should be printed according to the format specified in log4j2-test.json.

### Actual behavior
A list of errors is shown. Extract:
```
ERROR StatusLogger Unrecognized format specifier [d]
ERROR StatusLogger Unrecognized conversion specifier [d] starting at position 16 in conversion pattern.
ERROR StatusLogger Unrecognized format specifier [thread]
ERROR StatusLogger Unrecognized conversion specifier [thread] starting at position 25 in conversion pattern.
....
```
The exact behaviour depends on the setup:
#### Unit Test
In case of the local unit test, the custom logging configuration however is still applied.
#### Docker Container with custom lib within Neo4j plugins directory
In case of the docker container, the custom logging configuration is not applied. All log messages completely get lost. The different behaviour is probably related to order of classpath resource loading.

Side Note: The fix for #12631 seems okay as the following message disappeared with 4.2.1:
```
ERROR StatusLogger Log4j2 could not find a logging implementation. Please add log4j-core to the classpath. Using SimpleLogger to log to the console...
```
However the main issue as described above persists and completely blocks our upgrade to Neo4j 4.2.x currently.

**Comment by nioertel** [comment]
I also added a minimal reproduction case to this repo: https://github.com/nioertel/neo4j-log4j2

**Comment by emmaholmbergohlsson** [comment]
Thank you for reporting. We were able to reproduce the issue with the provided steps, and have found a fix. It will be included in the 4.2.4 version.
